### PR TITLE
[llvm] Use branch weight metadata in some cases.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by EgorBo** [comment]
@imhameed @vargaz  As far as I understand it applies some weight to the safepoint branches?
I thought we already use `llvm.expect` there which is converted into weights by `-lower-expect` pass.

**Comment by vargaz** [comment]
Some parts of llvm, not sure which don't know about llvm.expect so the safe point call is not moved out from inside the loop when using the llvm jit, with this branch metadata, it is moved out.

**Comment by EgorBo** [comment]
@vargaz um.. so `-lower-expect` (we probably need to make it the first pass) should just remove the `llvm.expect` completely and replace with !weight, example: https://godbolt.org/z/IS7Qqk

File: mono/mini/mini-llvm.c [hunk 1]
```diff
@@ -6345,7 +6345,7 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 			args [1] = LLVMConstInt (LLVMInt1Type (), 1, FALSE);
 			cmp = LLVMBuildCall (ctx->builder, get_intrins (ctx, INTRINS_EXPECT_I1), args, 2, "");
 
-			LLVMBuildCondBr (builder, cmp, cont_bb, poll_bb);
+			mono_llvm_build_weighted_branch (builder, cmp, cont_bb, poll_bb, 64, 4);
```
Comment (lewurm) [inline mono/mini/mini-llvm.c]: why 64 and 4?  is there a documentation for that somewhere?

Comment (vargaz) [reply mono/mini/mini-llvm.c]: @imhameed ^^

Comment (marek-safar) [reply mono/mini/mini-llvm.c]: I agree, make them constants at least

Comment (EgorBo) [reply mono/mini/mini-llvm.c]: In my sample https://godbolt.org/z/IS7Qqk `-lower-expect` used `1` and `2000` weights for branches.

Comment (EgorBo) [reply mono/mini/mini-llvm.c]: and I am actually not sure we use weights here in a correct order ("4" should be applied for the "we need to call state_pool" branch

Comment (vargaz) [reply mono/mini/mini-llvm.c]: cmp is the result of expect, so its expected to be 1.

**Pull request merged**